msb/analysis/timestepinterpolator.py

```diff
@@ -0,0 +1,51 @@
+import pandas as pd
+from msb.network.pubsub.types import Subscriber
+
+
+class PandasInterpolator:
+    """
+    An Iterator class that interpolates data points to a fixed rate.
+    """
+
+    def __init__(self, sub: Subscriber, freq_ms: float, sample_size: int = 100):
+        self.sub = sub
+        self.incoming: pd.DataFrame = None
+        self.outgoing: pd.DataFrame = None
+        self.resampling_frequency = freq_ms
+        self.sample_size = sample_size
+
+    def resample_loop(self) -> dict:
+        """
+        Generator loop that yields dicts at a fixed rate
+        """
+        while True:
+            self._listen_to_sub()
+            self._convert_timestamp()
```

Comment:
Consider renaming to e.g., `_epoch_to_datetime`.  As below, we convert the other way as well.